### PR TITLE
Proposal to change the last use section

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

File: DIPs/13NNN-WGB.md [hunk 1]
```diff
@@ -428,12 +428,92 @@ Class objects cannot have move constructors or move assignment operators.
 ### Last Use
 
 The *Last Use* of an lvalue is the last time in a path of execution that any read
-or write is done to the memory region referred to directly by the lvalue. For example:
+or write is done to the memory region referred to directly by the lvalue.
 
+**Return Statements**. If an lvalue is returned directly via `return x`, that is the last use of `x`.
```
Comment (andralex) [inline DIPs/13NNN-WGB.md]: what if `x` is a global?

File: DIPs/13NNN-WGB.md [hunk 2]
```diff
@@ -428,12 +428,92 @@ Class objects cannot have move constructors or move assignment operators.
...
+If the return statement is more complex, the rightmost use of the lvalue will be considered the last
```
Comment (andralex) [inline DIPs/13NNN-WGB.md]: s/is more complex/is an expression using a local `x` several times/

I guess

Comment (andralex) [reply DIPs/13NNN-WGB.md]: Then please mind really odd cases such as:

```d
return (() { ... use x ... })();
```

File: DIPs/13NNN-WGB.md (hunk 1 above)
Comment (WalterBright) [reply DIPs/13NNN-WGB.md]: There's never a last use of global. Only locals and parameters are checked.

File: DIPs/13NNN-WGB.md (hunk 2 above)
Comment (WalterBright) [reply DIPs/13NNN-WGB.md]: That falls under how references to outer variables in nested functions works.

**Pull request merged**